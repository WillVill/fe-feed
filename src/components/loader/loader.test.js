import React from 'react';
import Loader from './loader';
// Importing shallow and toJson.
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const setup = propOverrides => {
	const props = Object.assign({
		children: 'loading...',
		size: 'large',
		active: true
	}, propOverrides);

	const wrapper = shallow(<Loader {...props} />);

	return {
		props,
		wrapper
	};
};

describe('[Component] Loader', () => {
	it('renders', () => {
		const { wrapper } = setup();

		expect(toJson(wrapper)).toMatchSnapshot();
	});
});

it('shows proper loading text', () => {
	const { wrapper } = setup();

	expect(toJson(wrapper)).toMatchSnapshot();
});

