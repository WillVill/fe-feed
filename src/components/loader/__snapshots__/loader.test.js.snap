// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Loader renders 1`] = `
<div
  className="j-loader j-loader--large j-loader--true"
>
  <div
    className="j-loader__spinner"
  />
  <span
    className="j-loader__text"
  >
    loading...
  </span>
</div>
`;
